core/variant/array.cpp

```diff
@@ -494,9 +494,9 @@ void Array::set(int p_idx, const Variant &p_value) {
 	operator[](p_idx) = value;
 }
 
-const Variant &Array::get(int p_idx) const {
-	ERR_FAIL_INDEX_MSG(p_idx, size(), Variant());
-	return operator[](p_idx);
+Variant Array::get(int p_idx) const {
+    ERR_FAIL_INDEX_V_MSG(p_idx < 0 || p_idx >= size(), Variant());
```

Comment:
This chck is not required because `operator[]` already does the check. You'd do it twice if you have it here, too :)